### ISSUE TITLE
Order example gallery by popularity

### ISSUE THREAD

**Issue by bsipocz** [executablebooks/meta#369]
I would find it rather useful if the gallery is (roughly) ordered by the popularity of the project rather than the current random shuffle for each deployment (random by page view would be OK but random by build feels a bit weird).

Currently, it seems that many low traffic projects can end up at the top, and the most popular and probably mature ones (scikit-learn course, uw data viz, turning way) are towards the bottom.

https://executablebooks.org/en/latest/gallery.html

**Comment by chrisjsewell** [comment]
Certainly open to having some kind of option button to reorder by popularity "button", and yeh "order-by-view", although both are a bit difficult with the current generation method being static page creation. I think this will be more of a priority if/when the gallery grows to be ~>100 projects
 and it gets more difficult to browse the whole list.

I'm against popularity ordering being the default though, the reason being that the gallery is not just a showcase for "the popular projects"; anyone who takes the time to contribute should be given fair treatment for their project to be viewed. Ordering by popularity will just produce a horrible feedback loop of the popular projects becoming more popular and the smaller projects being ignored.

> Currently, it seems that many low traffic projects can end up at the top, and the most popular and probably mature ones (scikit-learn course, uw data viz, turning way) are towards the bottom.

I think "it seems" would be the operative word there lol, given that its completely random 😜 

 Note also the docs build is triggered every 24 hours, so the order is refreshed at a good rate.

**Comment by bsipocz** [comment]
> anyone who takes the time to contribute should be given fair treatment for their project to be viewed. Ordering by popularity will just produce a horrible feedback loop of the popular projects becoming more popular and the smaller projects being ignored.

I understand the sentiment, but I, as a new user when looking for examples to see how others deploy their book, am really interested in larger, established projects when shopping around for customisation examples that may be useful for my project, rather than spending much time in evaluating personal or very few contributor sized projects.

**Comment by jstac** [comment]
I was going to suggest listing the number of GH stars on the repo until I realized they were already on there...

That gives some guidance on popularity -- perhaps enough for the moment?

+1 for multiple viewing options (by date, GH stars, etc.) once the gallery hits 100.

**Comment by choldgraf** [comment]
I wonder if one option would be to use a table rather than a bunch of thumbnails, similar to what we have here:

https://executablebooks.org/en/latest/feature-vote.html

then one of the columns could be "stars" and people could easily sort by that if they wished.

**Comment by chrisjsewell** [comment]
> I wonder if one option would be to use a table rather than a bunch of thumbnails

Well then it wouldn't be a gallery, it would be a list lol (and not really very visually appealing). It really depends what people want to get out of it: for people building books, yes they may what to look for exemplars to get ideas from (which popularity can indicate, but is not necessarily always the case), but for people adding to the gallery it is a chance for them to show of their work. We have to balance these two.

Additionally, if only the most popular are promoted, then people will probably just stop feeling motivated to add their projects.

**Comment by choldgraf** [comment]
Yeah I agree it's not as visually appealing, though I think you can include images in datatables (e.g., [see this fiddle](http://jsfiddle.net/o1a3hyr6/3/)). So we could still retain the logo and some of the nice visual styling even if it wasn't strictly a table.

In this case, we could still randomize the entries of the table in the same way do now, it's just be easier for somebody to "sort" the table if they wanted to.

Another approach we could take is to do a hybrid of the two. Use the "gallery" to have, say "10 randomly selected books to promote" and underneath it a table of *all* books that have been submitted

**Comment by damianavila** [comment]
> Another approach we could take is to do a hybrid of the two. Use the "gallery" to have, say "10 randomly selected books to promote" and underneath it a table of all books that have been submitted

I like that idea, you are still getting the "gallery visual" first and more content in a condensed table if you are interested to find the most popular one (or any other filters).

**Comment by chrisjsewell** [comment]
Yeh I mean the easiest way is just to have a (sortable) list on a seperate page, linked to from the gallery
